src/twisted/trial/test/test_deferred.py

```diff
@@ -210,6 +210,19 @@ def test_timeoutZero(self) -> None:
         assert isinstance(result.errors[0][1], Failure)
         self._wasTimeout(result.errors[0][1])
 
+    def test_addCleanupPassDefault(self) -> None:
+        result = self.runTest("test_addCleanupPassDefault")
+        self.assertTrue(result.wasSuccessful())
+        self.assertEqual(result.testsRun, 1)
+
+    def test_addCleanupTimeout(self) -> None:
+        result = self.runTest("test_addCleanupTimeout")
```

Comment:
Maybe we can have it written like this

with a docstring and a test-double scheduler.

```suggestion
        """
        See L{twisted.trial.test.detests.TimeoutTests.test_addCleanupTimeout}
        """
        clock = task.Clock()
        self._twistedPrivateReactor = clock

        result = self.runTest("test_addCleanupTimeout")

        clock.advance(DEFAULT_TIMEOUT)
```

Reply:
Unfortunately this is not possible because runTest() eventually calls into TestCase._runFixturesAndTest where it waits via TestCase._wait() (which runs reactor inside) until the test either fails or succeeds. It would require quite a lot of refactoring to support fake reactor in this test.